### PR TITLE
refactor: consolidate/optimise grpc subscription setup code

### LINKED ISSUE #2166: Reduce frequency of getInfo pollng
## Description

We poll getInfo in order to determine the current sync state of lnd. We poll every 5 seconds. Once we know that the chain has been synced we can reduce this frequency down to something more like once per minute.

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

- disconnect getinfo stream on chain sync complete
- consolidate grpc subscription setup code
- unsubscribe streams before disconnecting
- dynamic gRPC subscription registration

## Motivation and Context:

Fix #2166 

Since we are not making use of the info subscription after the initial sync has completed I have opted to disable it once the sync has completed and we have activated the channel subscription. 

## How Has This Been Tested?

Manually

## Types of changes:

Refactor / fix

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: services/grpc/grpc.js [hunk 1]
```diff
@@ -19,15 +19,15 @@ class ZapGrpc extends EventEmitter {
...
-    'subscribeChannelGraph',
-    'subscribeTransactions',
-    'subscribeGetInfo',
-  ]
+  static SUBSCRIPTIONS = {
+    invoices: 'subscribeInvoices',
+    transactions: 'subscribeChannelGraph',
+    channelgraph: 'subscribeTransactions',
+    info: 'subscribeGetInfo',
+  }
```
Comment (korhaliv) [inline services/grpc/grpc.js]: Just a thought. Do we really need this to be hardcoded? I'm thinking we can use a more dynamic approach. like `registerService` and pre-register those 4. This eventually should work better with LND API versioning. Like for instances soon we will be adding `scb` to a list of subscriptions.

File: services/grpc/grpc.js [hunk 2]
```diff
@@ -112,47 +98,62 @@ class ZapGrpc extends EventEmitter {
...
     // this is needed because LND chanRouter waits for chain sync
-    // to complete before accepting subscriptions
+    // to complete before accepting subscriptions.
     this.on('subscribeGetInfo.data', data => {
       const { synced_to_chain } = data
-      if (synced_to_chain && !this.subscriptions['channelGraph']) {
-        const { Lightning } = this.services
-        this.subscriptions['channelGraph'] = Lightning.subscribeChannelGraph()
-        this.subscribe('channelGraph')
+      if (synced_to_chain && !this.subscriptions['channelgraph']) {
```
Comment (korhaliv) [inline services/grpc/grpc.js]: let's use `this.subscriptions.channelgraph` ?

Comment (mrfelton) [reply services/grpc/grpc.js]: Updated

File: services/grpc/grpc.js (hunk 1 above)
Comment (mrfelton) [reply services/grpc/grpc.js]: I've added a basic `registerSubscription` method that can be used to setup the mapping in a slightly more dynamic way.

**Pull request merged**